1_constants_variables_and_types/main.cpp

```diff
@@ -0,0 +1,395 @@
+#include <iostream>
+
+using namespace std;
+
+int iglob;
+static int isglob;
+const int icglob{123};
+const static int icsglob{234};
+auto aglob{345};
+
+double dglob;
+static double dsglob;
+const double dcglob{12.3};
+const static double dcsglob{23.4};
+auto adglob{34.5};
+
+// The function doesn't work because global variables sent with copy of value
+// template <typename T> inline void print_vars(string name, T  var, T * pvar, T & rvar)
```

Comment:
Done. The source code is changed to use template function.

Reply:
` std::cout << prefix << "Address: " << &var << "\n"; ` doesn't work because the compiler interpret `&char` variable as c-style string and tries to print whole string instead of address of pointer. To print address of char pointer we need to use `static_cast<const void *>`.